test/models/court_test.rb

```
@@ -1,7 +1,83 @@
 require 'test_helper'
 
 class CourtTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  def setup
+    @court = courts(:sample_court)
+  end
+
+  test 'should be valid' do
+    assert @court.save
+  end
+
+  test 'name should be present' do
+    @court.name = '   '
+    assert_not @court.valid?
```

Refute instead of assert_not would be better, furthermore we should always check wether the errors are the ones we expect here, this would be required for all validator tests.

---

I consider the error message as implementation detail, and thus not good practice to check on that. Furthermore, this is not necessary, as the setup-ed value has only valid entries - checked by `'should be valid'` -, and we are only changing selected entries to let the validation fail, so the cause of the failing validation is known  without checking on error strings.

---

We should always check what the errors are, it could be multiple validators triggering, maybe just one, maybe not the one we think is supposed to trigger here which might lead to confusing messages

+ it tests internationalization.